tests/sentry/eventstream/kafka/test_consumer.py

```diff
@@ -0,0 +1,632 @@
+        )
+
+        assignments_received = []
+
+        def on_assign(c, assignment):
+            assert c is consumer
+            assignments_received.append(assignment)
+
+        consumer.subscribe([topic], on_assign=on_assign)
+
+        # Wait until we have received our assignments.
+        for i in xrange(10):  # this takes a while
+            assert consumer.poll(1) is None
+            if assignments_received:
+                break
+
+        assert len(assignments_received) == 1, 'expected to receive partition assignment'
+        assert set((i.topic, i.partition) for i in assignments_received[0]) == set([(topic, 0)])
+
+        # TODO: Make sure that all partitions remain paused.
+
+        # Make sure that there are no messages ready to consume.
+        assert consumer.poll(1) is None
+
+        # Move the committed offset forward for our synchronizing group.
+        message = messages_delivered[topic][0]
+        producer.produce(
+            commit_log_topic,
+            key='{}:{}:{}'.format(
+                message.topic(),
+                message.partition(),
+                synchronize_commit_group,
+            ).encode('utf8'),
+            value='{}'.format(
+                message.offset() + 1,
+            ).encode('utf8'),
+        )
+
+        assert producer.flush(5) == 0, 'producer did not successfully flush queue'
+
+        # We should have received a single message.
+        # TODO: Can we also assert that the position is unpaused?)
+        for i in xrange(5):
+            message = consumer.poll(1)
+            if message is not None:
+                break
+
+        assert message is not None, 'no message received'
+
+        expected_message = messages_delivered[topic][0]
+        assert message.topic() == expected_message.topic()
+        assert message.partition() == expected_message.partition()
+        assert message.offset() == expected_message.offset()
+
+        # We should not be able to continue reading into the topic.
+        # TODO: Can we assert that the position is paused?
+        assert consumer.poll(1) is None
+
+
+@requires_kafka_client
+def test_consumer_start_from_committed_offset():
+    consumer_group = 'consumer-{}'.format(uuid.uuid1().hex)
+    synchronize_commit_group = 'consumer-{}'.format(uuid.uuid1().hex)
+
+    messages_delivered = defaultdict(list)
+
+    def record_message_delivered(error, message):
+        assert error is None
+        messages_delivered[message.topic()].append(message)
+
+    producer = Producer({
+        'bootstrap.servers': os.environ['SENTRY_KAFKA_HOSTS'],
+        'on_delivery': record_message_delivered,
+    })
+
+    with create_topic() as topic, create_topic() as commit_log_topic:
+
+        # Produce some messages into the topic.
+        for i in range(3):
+            producer.produce(topic, '{}'.format(i).encode('utf8'))
+
+        assert producer.flush(5) == 0, 'producer did not successfully flush queue'
+
+        Consumer({
```

Comment:
I'm a little bit unclear on why this block is needed. If the `SynchronizedConsumer` checks the progress based on the `commit_log_topic`,  then `commit()`ing the offsets to `topic` here as well should have no effect? On the other hand, I guess it keeps the world nice and consistent to make sure that the offsets committed to `topic` and the messages sent to `commit_log_topic` are lined up properly. 

 If it is needed, I don't quite understand why we do it for the first message and not the second.

Reply:
I think (since this is committing to `consumer_group` and not `synchronize_commit_group`) that I was trying to ensure that if the consumer offset advances beyond the one in the commit log for some reason (e.g. the position written to the commit log moves backwards) that the synchronized consumer does not have its own offset rolled back?

Reply:
ah yeah that makes sense. I was imagining this consumer to be posting its regular kafka consumer offsets to `synchronize_commit_group`, which I assume would have no effect on the SynchronizedConsumer